packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -189,14 +171,95 @@ export class Builder extends EventEmitter {
     await this.generateDocumentTypes();
   }
 
-  private async generateSchemaTypes(schemaPath: string, schema: GraphQLSchema) {
-    const schemaTypesPath = getSchemaTypesPath(
-      getGraphQLProjectForSchemaPath(this.config, schemaPath),
-      this.options,
+  stop() {
+    this.watchers.forEach((watcher) => {
+      watcher.close();
+    });
+
+    this.watchers.length = 0;
+  }
+
+  private setupDocumentWatchers() {
+    const update = (filePath: string, projectConfig: GraphQLProjectConfig) => {
+      try {
+        this.updateDocumentForFileSync(filePath, projectConfig);
```

Comment:
So, I am confused by this change. You reference #28 in your PR description as to why we move to sync functions for these callbacks, but it seems that just the `ignoreInitial` change addresses that. I worry a lot about doing all of the synchronous operations since this whole thing happens alongside other expensive operations in Sewing Kit.

Reply:
the `ignoreInitial` does fix the specific problem that #28 represents, but it does not fix the fact that we cannot _safely_ run async code in the `chokidar` event handlers (The reference to #28 was just that part of that effort resolves it, the remainder of the effort resolves the async issues). These sync calls in the `update` function ought to be limited to actual file changes, so in general we should not be hitting this function too often. All of the _spin up_ processing should be handled by the async `run` function which is offloading all processing onto async functions.